test/e2e/router_compression_test.go

```diff
@@ -0,0 +1,261 @@
+		t.Errorf(fmt.Sprintf(errorMsg, err))
+	}
+	defer assertIngressControllerDeleted(t, kclient, pc)
+}
+
+func getPrivateController(t *testing.T, privateName string, privateDomain string) (*operatorv1.IngressController, error) {
+	t.Helper()
+
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: privateName}
+	domain := icName.Name + "." + privateDomain
+
+	ic := newPrivateController(icName, domain)
+	// Create a new private Ingress Controller (deletion handled by caller)
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		return ic, errors.New(fmt.Sprintf("error creating private ingresscontroller %s: %v", privateName, err))
+	}
+	return ic, nil
+}
+
+// testCompressionPolicy updates the given ingresscontroller's spec.httpCompressionPolicy with the given
+// compressionPolicy, and reports an error if it the compression policy was not accepted or not added to the router
+// deployment environment.
+func testCompressionPolicy(t *testing.T, name string, compressionPolicy operatorv1.HTTPCompressionPolicy) error {
+	t.Helper()
+	namespacedName := types.NamespacedName{Namespace: operatorNamespace, Name: name}
+	routerDeploymentNamespacedName := types.NamespacedName{Namespace: controller.DefaultOperandNamespace, Name: "router-" + name}
+
+	if err := wait.PollImmediate(10*time.Second, 2*time.Minute, func() (bool, error) {
+		ic, err := getIngressController(t, kclient, namespacedName, 5*time.Second)
+		if err != nil {
+			t.Logf("failed to get ingress controller: %v, retrying...", err)
+			return false, nil
+		}
+		ic.Spec.HTTPCompression = compressionPolicy
+
+		if err := kclient.Update(context.TODO(), ic); err != nil {
+			if strings.Contains(err.Error(), "has been modified") {
+				t.Logf("failed to update ingress controller, retrying...")
+				return false, nil
+			}
+			// Return an error if validation failed
+			return true, err
+		}
+		return true, nil
+	}); err != nil {
+		return errors.New(fmt.Sprintf("failed to update ingress controller: %v", err))
+	}
+
+	// Now wait for it to become available
+	conditions := []operatorv1.OperatorCondition{
+		{Type: operatorv1.IngressControllerAvailableConditionType, Status: operatorv1.ConditionTrue},
+	}
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, namespacedName, conditions...); err != nil {
+		t.Errorf("failed to observe expected conditions: %v", err)
+		return err
+	}
+
+	// Get the router deployment
+	deployment, err := getDeployment(t, kclient, routerDeploymentNamespacedName, 2*time.Minute)
+	if err != nil {
+		t.Errorf("failed to get deployment: %v", err)
+		return err
+	}
+
+	// Check if the MIME type environment variable has been updated
+	mimeTypes := ingress.GetMIMETypes(compressionPolicy.MimeTypes)
+	if err := waitForDeploymentEnvVar(t, kclient, deployment, 2*time.Minute, "ROUTER_COMPRESSION_MIME", strings.Join(mimeTypes, " ")); err != nil {
+		t.Errorf("expected deployment to have mimeTypes %s: %v", mimeTypes, err)
+		return err
```

Comment:
Can `t.Errorf` be deleted and a explanatory error value be returned?  
```suggestion
		return fmt.Errorf("expected deployment to have mimeTypes %q: %w", mimeTypes, err)
```